beacon-chain/blockchain/pow_block.go

```diff
@@ -92,6 +93,7 @@ func (s *Service) getBlkParentHashAndTD(ctx context.Context, blkHash []byte) ([]
 	if blk == nil {
 		return nil, nil, errors.New("pow block is nil")
 	}
+	blk.Version = version.Bellatrix
```

Comment:
Why set this ? Shouldn't it already have its version set in the `ExecutionBlockByHash` method ? Or are you not able to infer that

Reply:
Not able to infer from how `ExecutionBlockByHash` is implemented. I can add a `version` argument, but this seems cleaner to me.